realm/realm-library/src/main/java/io/realm/internal/NativeObjectReference.java

```diff
@@ -22,13 +22,21 @@
  * This class is used for holding the reference to the native pointers present in NativeObjects.
  * This is required as phantom references cannot access the original objects for this value.
  */
-public class NativeObjectReference extends PhantomReference<NativeObject> {
+public abstract class NativeObjectReference extends PhantomReference<NativeObject> {
 
     // The pointer to the native object to be handled
-    final long nativePointer;
+    protected final long nativePointer;
 
     public NativeObjectReference(NativeObject referent, ReferenceQueue<? super NativeObject> referenceQueue) {
         super(referent, referenceQueue);
         nativePointer = referent.nativePointer;
     }
+
+    protected abstract void cleanup();
+
+    @Override
+    public void clear() {
+        cleanup();
```

Comment:
`clear` will set `referent` to null. Maybe the `cleanup` wants to do something with the `referent`. Although it is not necessary right now.